internal/remoteconfig/remoteconfig.go

```diff
@@ -95,34 +103,60 @@ func NewClient(config ClientConfig) (*Client, error) {
 	}, nil
 }
 
-// Start starts the client's update poll loop in a fresh goroutine
-func (c *Client) Start() {
-	go func() {
-		ticker := time.NewTicker(c.PollInterval)
-		defer ticker.Stop()
-
-		for {
-			select {
-			case <-c.stop:
-				close(c.stop)
-				return
-			case <-ticker.C:
-				c.updateState()
+// Start starts the client's update poll loop in a fresh goroutine.
+// Noop if the client has already started.
```

Comment:
Listened products and tracer capabilities must be merged with the already instantiated one and not ignored here.

Reply:
Shouldn't products call `RegisterProduct` and `RegisterCallback` to register their products and callbacks? 🤔 
Is there real value in keeping `Products` and `Capabilities` public as part of the `ClientConfig` ? I'd prefer to have a unique and simple way to register Products and Capabilities but lmk what you think!

Reply:
>BTW, another alternative could be removing these shared config fields from the struct and replace them by singleton functions only, so that there is no more ambiguity on the Start() config.

Agreed, didn't see your second message when I wrote my comment!

Reply:
My bad I didn't realize we weren't using those fields neither, so this is great news and worth the struct clean up, thanks :-)